hal/src/nRF52840/exflash_hal.c

```diff
@@ -571,6 +542,48 @@ int hal_exflash_special_command(hal_exflash_special_sector_t sp, hal_exflash_com
         }
     }
 
+exit:
     hal_exflash_unlock();
     return ret;
 }
+
+int hal_exflash_sleep(bool sleep, void* reserved) {
+    hal_exflash_lock();
+    if (sleep) {
+        // Suspend external flash
+        if (qspi_state != HAL_EXFLASH_STATE_ENABLED) {
+            hal_exflash_unlock();
+            return SYSTEM_ERROR_INVALID_STATE;
+        }
+        // Put external flash into Deep Power-down mode
+        if (hal_exflash_special_command(HAL_EXFLASH_SPECIAL_SECTOR_NONE, HAL_EXFLASH_COMMAND_SLEEP, NULL, NULL, 0)) {
+            // It may fail as the flash chip might be in an ongoing program/erase operation. Suspend it.
+            hal_exflash_special_command(HAL_EXFLASH_SPECIAL_SECTOR_NONE, HAL_EXFLASH_COMMAND_SUSPEND_PGMERS, NULL, NULL, 0);
+            // Abort the suspended write/erase operation.
+            hal_exflash_special_command(HAL_EXFLASH_SPECIAL_SECTOR_NONE, HAL_EXFLASH_COMMAND_RESET, NULL, NULL, 0);
```

Comment:
Just one minor thing: after this command you might need to send `QSPI_STD_CMD_WRSR`. Double check the datasheet.